pkg/operator/status.go

```diff
@@ -23,6 +23,15 @@ const (
 	ReasonCheckAutoscaler   = "UnableToCheckAutoscalers"
 )
 
+// StatusReporterInterface is used to enable unit testing and does not
+// implement all methods of StatusReporter.
+type StatusReporterInterface interface {
```

Comment:
I find the name confusing as we already have `StatusReporter`. Can we differentiate with another name?